### PR TITLE
ENH: allow passing MAGs to the compute action

### PR DESCRIPTION
**Pull request by misialq**
This PR adds an ability to pass MAG sequences to the `compute` action, so that the MinHash signatures can also be computed for this kind of DNA sequences.

### CODE REVIEW

**Comment by misialq** [comment]
Hey @ctb, these are some changes that @nbokulich mentioned to you - do you think you could have a look? 🙏

File: q2_sourmash/_compute.py [hunk 1]
```diff
@@ -5,40 +5,81 @@
...
-        # print(sampleid, filename)
-        qiime2.util.duplicate(filepath, os.path.join(str(output), sampleid[0]+'.fastq.gz'))
+    if isinstance(sequence_file, MultiMAGSequencesDirFmt):
+        output = _duplicate_mag_seqs(manifest, sequence_file)
+    else:
+        output = _duplicate_seqs(manifest, sequence_file)
 
-    command = ['sourmash', 'compute', str(output) + "/*", '--ksizes', str(ksizes), '--scaled', str(scaled)]
+    command = [
+        'sourmash', 'compute', f'{output}/*',
```
Comment (ctb) [inline q2_sourmash/_compute.py:67]: in our next major release of sourmash, sourmash v5, we will be removing `sourmash compute` per https://github.com/sourmash-bio/sourmash/issues/1286.

It's been replaced with sourmash sketch; the following should work:
```
ksizestr = [ f"k={k}" for k in ksizes ]
ksizestr = ",".join(ksizestr)
abundstr = "noabund"
if track_abundance:
   abundstr = "abund"

command = ['sourmash', 'sketch', 'dna', f'{output}/*', '-p', f'scaled={scaled},{ksizestr},{abundstr}'
```

**Comment by ctb** [comment]
Commented in one place - and overall looks good to me. Should I approve in review & merge?

File: q2_sourmash/_compute.py (hunk 1 above)
Comment (misialq) [reply q2_sourmash/_compute.py:67]: Good to know, thanks for info! Should we then update it in a separate PR, once v5 is out?

**Comment by misialq** [comment]
That would be lovely, thanks!

**Pull request merged**